ibis/expr/tests/test_rules.py

```diff
@@ -170,3 +170,184 @@ class MyOp(ops.ValueOp):
 
     op = MyOp(True)
     assert op.value.equals(ibis.literal(True))
+
+
+def check_op_input(Op, schema, raises):
+    schema = ibis.Schema.from_tuples(schema)
+    table = ibis.table(schema)
+    if not raises:
+        assert Op(table).table.equals(table)
+    else:
+        with pytest.raises(IbisTypeError):
+            Op(table)
+
+
+@pytest.mark.parametrize(
```

Comment:
I tried that and it was a mess, because the same schemas would be repeated twice, once for tests passing validation and those not passing.

I think it is waaay clearer to say "given this schema, here are things that pass and things that don't pass".

Reply:
Sorry, didn't explain this right. Indeed, the schemas for the tables being validated would not be repeated. Bu the schema we are comparing against will be repeated.

Reply:
This would end up looking as:
```
@pytest.mark.parametrize(
    "... many passing test cases here ...")
def test_table_with_schema():
    "<validation code with a schema>"

@pytest.mark.parametrize(
    "... failing test cases here ...")
def test_table_with_schema_failing():
    with pytest.raises(IbisTypeError):
        "<same validation code with the same schema>"
```

Reply:
Eh, did it like that. Its not so bad.